src/server_manager/ui_components/outline-manual-server-entry.html

```diff
@@ -246,119 +360,193 @@
+          >
         </div>
-        <paper-progress hidden$='[[!showConnection]]' indeterminate class="slow"></paper-progress>
+        <paper-progress
+          hidden$="[[!showConnection]]"
+          indeterminate
+          class="slow"
+        ></paper-progress>
       </div>
     </outline-step-view>
   </template>
   <script>
     Polymer({
-      is: 'outline-manual-server-entry',
+      is: "outline-manual-server-entry",
       properties: {
+        installScriptStepNumber: {
+          type: Number,
+          value: 2
+        },
+        pasteJsonStepNumber: {
+          type: Number,
+          value: 3
+        },
         placeholderText: {
           type: String,
-          value: '{"apiUrl":"https://xxx.xxx.xxx.xxx:xxxxx/xxxxxxxxxxxxxxxxxxxxxx","certSha256":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"}'
-
+          value:
+            '{"apiUrl":"https://xxx.xxx.xxx.xxx:xxxxx/xxxxxxxxxxxxxxxxxxxxxx","certSha256":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"}'
         },
         showConnection: Boolean,
         cloudProvider: {
           type: String,
-          value: 'generic'
+          value: "generic",
+          observer: "_cloudProviderChanged"
         },
         cloudProviderName: {
           type: String,
-          computed: '_computeCloudProviderName(cloudProvider)'
+          computed: "_computeCloudProviderName(cloudProvider)"
         },
         isCloudProviderAws: {
           type: Boolean,
-          computed: '_computeIsCloudProviderAws(cloudProvider)'
+          computed: "_computeIsCloudProviderAws(cloudProvider)"
         },
         isCloudProviderGcp: {
           type: Boolean,
-          computed: '_computeIsCloudProviderGcp(cloudProvider)'
+          computed: "_computeIsCloudProviderGcp(cloudProvider)"
         },
         isGenericCloudProvider: {
           type: Boolean,
-          computed: '_computeIsGenericCloudProvider(cloudProvider)'
+          computed: "_computeIsGenericCloudProvider(cloudProvider)"
         },
         enableDoneButton: {
           type: Boolean,
-          value: false,
+          value: false
         },
         localize: {
           type: Function,
-          readonly: true,
-        },
+          readonly: true
+        }
       },
       doneTapped: function() {
         this.showConnection = true;
-        this.fire('ManualServerEntered', {
+        this.fire("ManualServerEntered", {
           userInput: this.$.serverConfig.value
         });
       },
       cancelTapped: function() {
-        this.fire('ManualServerEntryCancelled');
+        this.fire("ManualServerEntryCancelled");
       },
       retryTapped: function() {
         this.showConnection = false;
         this.doneTapped();
       },
       clear: function() {
-        this.$.serverConfig.value = '';
+        this.$.serverConfig.value = "";
         this.showConnection = false;
         this.$.awsDropDown.hide();
```

Comment:
You want to compute the opened property of the different collapses when the user clicks?  It seems like not much gain for a larger refactor.  To be honest I'm not even sure why we want them to be collapsed like this, I don't see an issue with the instructions being in the same place as they were before.

Reply:
Ok, we don't need to change it. But the existing code is too hard to read. We are not really super experienced web developers and used some patterns that aren't really good practices. For example, writing our own event handling to update data instead of using data binding. As long as we are aware of those patterns and don't repeat it's ok not changing.